src/blob_file_set.cc

```diff
@@ -136,8 +131,8 @@ Status BlobFileSet::Recover() {
                    "Titan recovery delete obsolete file %s.", f.c_str());
     env_->DeleteFile(dirname_ + "/" + f);
   }
-
-  return Status::OK();
+  auto new_manifest_file_number = NewFileNumber();
+  return OpenManifest(new_manifest_file_number);
```

Comment:
do not change the order, otherwise the old manifest would be already deleted when failed to create new manifest

Reply:
PTAL